liberty-maven-plugin/src/main/java/net/wasdev/wlp/maven/plugins/server/InstallFeatureMojo.java

```diff
@@ -47,12 +82,262 @@ protected void doExecute() throws Exception {
+                log.debug("Server XML feature: " + sf);
+            }
+        } else {
+            log.debug("Features were specified in configuration. Skipping server.xml.");
+            serverFeatures = new ArrayList<String>();
+        }
+        return serverFeatures;
+    }
+
+    private void populateConfigurationFeaturesLists(List<String> configuredFeatures, List<Feature> configuredEsas) {
+        for (Feature feature : features.getFeatures()) {
+            if (feature.getFeature().endsWith(".esa")) {
+                configuredEsas.add(feature);
+                log.debug("Configured ESA: " + feature);
+            } else {
+                configuredFeatures.add(feature.getFeature());
+                log.debug("Configured feature: " + feature.getFeature());
+            }
+        }
+    }
+
+    private Set<File> downloadProductJsons() throws MojoExecutionException {
+        // get productId and version for all properties
+        File versionsDir = new File(installDirectory, "lib/versions");
+        List<String[]> propertiesList = loadProperties(versionsDir);
+
+        if (propertiesList.isEmpty()) {
+            throw new MojoExecutionException("Could not find any properties file in the " + versionsDir
+                    + " directory. Ensure the directory " + installDirectory + " contains a Liberty installation.");
+        }
         
+        // download JSONs
+        Set<File> downloadedJsons = new HashSet<File>();
+        for (String[] properties : propertiesList) {
+            List<File> jsons = downloadJsons(properties[0], properties[1]);
+            if (jsons != null) {
+                downloadedJsons.addAll(jsons);
+            }
+        }
+        return downloadedJsons;
+    }
+
+    private void handleJsonsNotFound() {
+        log.debug("No jsons found corresponding to the runtime's productId");
+        if (hasInstallUtility()) {
+            log.debug("Installing features from installUtility");
+            installFeaturesFromAnt(features.getFeatures());
+        } else {
+            log.info("Feature installation is not supported on this version of Open Liberty");
+        }
+    }
+    
+    /**
+     * Fall back to installUtility implementation if one of these conditions are
+     * true: jar does not exist, "from" parameter is specified (don't need
+     * Maven repositories), "to" parameter is specified and not usr/core (only
+     * usr/core are supported with Maven for now), or esa files are specified in
+     * the configuration (not supported with Maven for now)
+     * 
+     * @param configuredEsas
+     * @return true if the fallback scenario occurred, false otherwise
+     * @throws MojoExecutionException
+     */
+    private boolean handleFallbackScenarios(List<Feature> configuredEsas) throws MojoExecutionException {
+
+        boolean hasInstallKernelJar = getMapBasedInstallKernelJar() != null;
+        boolean hasFrom = features.getFrom() != null;
```

Comment:
Those are needed to check for parameters that are not currently supported by the install map jar.  Moved to ci.common InstallFeatureUtil.hasUnsupportedParameters().

Reply:
When you say "not currently supported", does that mean that these will be supported in future versions? If so, how will this work code work in the future? Is there a way we can inspect the install map JAR to see which parameters are supported instead of hard-coding this?

Reply:
The install map's interface does not support those options.  The "from" parameter will not be added since it should install from a specific repository (not Maven).  The "to" and esa parameters will be added with new interfaces/parameters that have not been defined yet, so the check should occur in the plugin and this check is all contained in one method.

Reply:
Ok. Once the updated version is released you will have to consider how to figure out what interfaces are supported dynamically.